r/R/dataset-write.R

```diff
@@ -123,7 +124,7 @@ write_dataset <- function(dataset,
                           path,
                           format = c("parquet", "feather", "arrow", "ipc", "csv"),
                           partitioning = dplyr::group_vars(dataset),
-                          basename_template = paste0("part-{i}.", as.character(format)),
+                          basename_template = NULL,
```

Comment:
I actually don't think you need this change due to the magic of lazy evaluation, you just need to add `format <- ifelse(as.character(format) %in% c("feather", "ipc"), "arrow", format)` after `format <- match.arg(format)`

```
> f <- function(x = c("a", "b", "c"), basename_template = paste0("part-{i}.", x)) {
+   x <- match.arg(x)
+   x <- toupper(x)
+   basename_template
+ }
> f()
[1] "part-{i}.A"
```

The benefit of this is that you still get to see the default value for `basename_template` in the signature/docs.

Reply:
I'm not familiar with lazy evaluation, and it's still difficult for me. I learned a lot. Thank you very much.

I did some updates.
please check it.